config/v2/config.go

```diff
@@ -152,12 +152,18 @@ type DatadogProvider struct{}
 
 //Backend is used to configure a terraform backend
 type Backend struct {
```

Comment:
random thought: I assume all the fields in this struct are left as is/at the top level to preserve backwards compatibility.
I wonder if we can add a notion of grouping by doing something like
```
Backend struct {
Kind *string
S3Backend //embedded
TFEBackend //embedded
```
that way we keep the same structure in the json but have some notion of semantic grouping in the code. not sure how I feel about this though, maybe a terrible idea.

Reply:
Interesting idea, but not sure what I think about it.